kuma/wiki/views/list.py

```diff
@@ -190,30 +189,28 @@ def get_previous(revisions):
 
     # Grab revisions, but defer summary and content because they can lead to
     # attempts to cache more than memcached allows.
-    revisions = MultiQuerySet(
-        (Revision.objects.filter(pk=document.current_revision.pk)
-                         .prefetch_related('creator', 'document')
-                         .transform(get_previous)),
-        (Revision.objects.filter(document=document)
-                         .order_by('-created', '-id')
-                         .exclude(pk=document.current_revision.pk)
-                         .prefetch_related('creator', 'document')
-                         .transform(get_previous))
-    )
-
-    if not revisions.exists():
+    all_revisions = (document.revisions.defer('summary', 'content').order_by('created', 'id')
+                                                                   .reverse().transform(get_previous))
+
+    if not all_revisions.exists():
         raise Http404
 
     if per_page == 'all':
         page = None
+        revisions = list(all_revisions)
     else:
         try:
             per_page = int(per_page)
         except ValueError:
             per_page = DOCUMENTS_PER_PAGE
 
-        page = paginate(request, revisions, per_page)
-        revisions = page.object_list
+        page = paginate(request, all_revisions, per_page)
+        revisions = list(page.object_list)
+    # In order to compare the first revision of a translation, need to insert its parent revision to the list
+    # The parent revision should stay at last page in order to compare. So insert only if there are no next page or
+    # all revisions are showing
+    if (not page or not page.has_next()) and document.parent:
+        revisions.append(all_revisions.last().based_on)
```

Comment:
Not all revisions have a ``based_on`` set.  123228 of the 261724 non-English translations in production (47%) do not.  It is possible none of the first translations do 😢.